src/agent.spec.ts

```diff
@@ -0,0 +1,157 @@
+import { FindingType, FindingSeverity, Finding, HandleTransaction } from "forta-agent";
+import { createAddress } from "forta-agent-tools";
+import { provideHandleTransaction } from "./agent";
+import { TestTransactionEvent } from "forta-agent-tools/lib/test";
+import { Interface } from "@ethersproject/abi";
+import { BigNumber } from "ethers";
+
+import { NETHERMIND_DEPLOYER_ADDRESS, CREATE_AGENT, FORTA_CONTRACT_ADDRESS } from "./utils";
+
+const TEST_DATA_1 = {
+  agentId: "44444444",
+  owner: NETHERMIND_DEPLOYER_ADDRESS,
+  chainIds: "333",
+  metadata: "abcdefghi",
+};
+const TEST_DATA_2 = {
+  agentId: "4444",
+  owner: createAddress("0x44"),
+  chainIds: "222",
+  metadata: "jklmnopqr",
+};
+
+const TEST_DEPLOYER_ADDRESS = createAddress("0x123");
+const TEST_DEPLOYER_ADDRESS2 = createAddress("0x456");
+
+describe("Nethermind Agent", () => {
+  let handleTransaction: HandleTransaction;
+  let fortaProxy = new Interface([CREATE_AGENT]);
+  let txEvent;
```

Comment:
it'd be better if `txEvent` is defined separately in each case as a const. here it doesn't affect things but for more complicated bots it's easier to have it separated for each scenario.